halo/attest/keeper/keeper.go

```diff
@@ -462,7 +462,46 @@ func (k *Keeper) latestAttestation(ctx context.Context, version xchain.ChainVers
 	}
 
 	if iter.Next() {
-		return nil, false, errors.New("multiple attestation found")
+		return nil, false, errors.New("multiple attestations" +
+			" found")
+	}
+
+	// If this attestation is overridden by a finalized attestation, return that instead.
+	if att.GetFinalizedAttId() != 0 {
+		att, err := k.attTable.Get(ctx, att.GetFinalizedAttId())
+		if err != nil {
+			return nil, false, errors.Wrap(err, "get finalized attestation")
+		}
+
+		return att, true, nil
+	}
+
+	return att, true, nil
+}
+
+// earliestAttestationInState returns the earliest approved attestation for the given chain currently found in state,
+// or false if none is found.
+func (k *Keeper) earliestAttestationInState(ctx context.Context, version xchain.ChainVersion) (*Attestation, bool, error) {
```

Comment:
Instead of adding a new method here, we could combine this into `latestAttestation` and save some query calls. However, I'm guessing what we _really_ want to conserve is database reads, and I think we'll still need two reads (one reversed and one not).